test/unittests/libp2p/net.cpp

```diff
@@ -1173,6 +1147,128 @@ BOOST_AUTO_TEST_CASE(addNodePingsNodeOnlyOnce)
+        nodeSocketHost1.start();
+        nodePort1 = nodeSocketHost1.port;
+        nodeSocketHost2.start();
+        nodePort2 = nodeSocketHost2.port;
+
+        nodeEndpoint1 =
+            NodeIPEndpoint{bi::address::from_string(c_localhostIp), nodePort1, nodePort1};
+        nodeEndpoint2 =
+            NodeIPEndpoint{bi::address::from_string(c_localhostIp), nodePort2, nodePort2};
+
+        // add a node to node table, initiating PING
+        nodeTable->addNode(Node{nodePubKey, nodeEndpoint1});
+
+        // handle received PING
+        auto pingDataReceived = nodeSocketHost1.packetsReceived.pop();
+        auto pingDatagram =
+            DiscoveryDatagram::interpretUDP(bi::udp::endpoint{}, dev::ref(pingDataReceived));
+        auto ping = dynamic_cast<PingNode const&>(*pingDatagram);
+
+        // send PONG
+        Pong pong(nodeTable->m_hostNodeEndpoint);
+        pong.echo = ping.echo;
+        pong.sign(nodeKeyPair.secret());
+        nodeSocketHost1.socket->send(pong);
+
+        // wait for PONG to be received and handled
+        nodeTable->packetsReceived.pop(chrono::seconds(5));
+
+        nodeEntry = nodeTable->nodeEntry(nodePubKey);
+    }
+
+    TestNodeTableHost nodeTableHost{0};
+    shared_ptr<TestNodeTable>& nodeTable = nodeTableHost.nodeTable;
+
+    // socket representing initial peer node
+    TestUDPSocketHost nodeSocketHost1;
+    uint16_t nodePort1 = 0;
+
+    // socket representing peer with changed endpoint
+    TestUDPSocketHost nodeSocketHost2;
+    uint16_t nodePort2 = 0;
+
+    NodeIPEndpoint nodeEndpoint1;
+    NodeIPEndpoint nodeEndpoint2;
+    KeyPair nodeKeyPair = KeyPair::create();
+    NodeID nodePubKey = nodeKeyPair.pub();
+
+    shared_ptr<NodeEntry> nodeEntry;
+};
+
+BOOST_FIXTURE_TEST_SUITE(packetsWithChangedEndpointSuite, PacketsWithChangedEndpointFixture)
+
+BOOST_AUTO_TEST_CASE(addNode)
+{
+    // this should initiate Ping to endpoint 2
+    nodeTable->addNode(Node{nodePubKey, nodeEndpoint2});
+
+    // handle received PING
+    auto pingDataReceived = nodeSocketHost2.packetsReceived.pop();
+    auto pingDatagram =
+        DiscoveryDatagram::interpretUDP(bi::udp::endpoint{}, dev::ref(pingDataReceived));
+    auto ping = dynamic_cast<PingNode const&>(*pingDatagram);
+
+    // send PONG
+    Pong pong(nodeTable->m_hostNodeEndpoint);
+    pong.echo = ping.echo;
+    pong.sign(nodeKeyPair.secret());
+    nodeSocketHost2.socket->send(pong);
+
+    // wait for PONG to be received and handled
+    nodeTable->packetsReceived.pop(chrono::seconds(5));
+
+    BOOST_REQUIRE_EQUAL(nodeEntry->endpoint, nodeEndpoint2);
+}
+
+BOOST_AUTO_TEST_CASE(findNode)
+{
+    // Create and send the FindNode packet through endpoint 2
+    FindNode findNode(nodeTable->m_hostNodeEndpoint, KeyPair::create().pub() /* target */);
+    findNode.sign(nodeKeyPair.secret());
+    nodeSocketHost2.socket->send(findNode);
+
+    // Wait for FindNode to be received
+    nodeTable->packetsReceived.pop(chrono::seconds(5));
```

Comment:
It might make sense to validate the name of the received packet to make the test easier to quickly debug in the event of a failure